packages/ui/src/helpers/storage/fileUploader/utils/uploader.ts

```diff
@@ -0,0 +1,96 @@
+import { Storage } from 'aws-amplify';
+import { StorageAccessLevel } from '@aws-amplify/storage';
+import { translate } from '../../../../i18n';
+
+export function uploadFile({
+  file,
+  fileName,
+  level = 'private',
+  progressCallback,
+  errorCallback,
+  completeCallback,
+  resumable = false,
+  ...rest
+}: {
+  file: File;
+  fileName: string;
+  level: StorageAccessLevel;
+  resumable?: boolean;
+  progressCallback: (progress: { loaded: number; total: number }) => void;
+  errorCallback: (err: string) => void;
+  completeCallback: (event) => void;
+}) {
+  if (resumable === true) {
+    return Storage.put(fileName, file, {
+      level,
+      resumable,
+      progressCallback,
+      errorCallback,
+      // TODO: Remove this override once we depend on aws-amplify@5
+      // this behavior is fixed in version 5
+      completeCallback: (event) => {
+        completeCallback({ key: event.key.split('/').slice(1).join('/') });
+      },
+      ...rest,
+    });
+  } else {
+    return Storage.put(fileName, file, {
+      level,
+      resumable: false,
+      progressCallback,
+      ...rest,
+    }).then(completeCallback, errorCallback);
+  }
+}
+/**
+ * Format bytes as human-readable text.
+ *
+ * @param bytes Number of bytes.
+ * @param si True to use metric (SI) units, aka powers of 1000. False to use
+ *           binary (IEC), aka powers of 1024.
+ * @param dp Number of decimal places to display.
+ *
+ * @return Formatted string.
+ */
+export function humanFileSize(bytes, si = false, dp = 1) {
+  const thresh = si ? 1000 : 1024;
+
+  if (Math.abs(bytes) < thresh) {
+    return bytes + ' B';
+  }
+
+  const units = si
+    ? ['kB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+    : ['KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB'];
+  let unit = -1;
+  const range = 10 ** dp;
+
+  do {
+    bytes /= thresh;
+    ++unit;
+  } while (
+    Math.round(Math.abs(bytes) * range) / range >= thresh &&
+    unit < units.length - 1
+  );
+
+  return bytes.toFixed(dp) + ' ' + units[unit];
+}
+
+export const checkMaxSize = (maxSize: number, file: File): string | null => {
+  if (!maxSize) return null;
+  if (file.size > maxSize) {
+    return translate('Size above max ') + humanFileSize(maxSize, true);
+  }
+  return null;
+};
+
+export const returnAcceptedFiles = (
+  files: File[],
+  acceptedFileTypes: string[]
+): File[] => {
+  // Remove any files that are not in the accepted file list
+  return [...files].filter((file) => {
```

Comment:
I did this because before sometimes `files` would be a `FileList` that can't be iterated through. But I checked the code we are using `File[]` everywhere so this should be OK. Will update

Reply:
[b3d449c](https://github.com/aws-amplify/amplify-ui/pull/3028/commits/b3d449cb5b4833d154b3c750b5c31275d2b25b25)